### ISSUE TITLE
Non-linearly through the dialog tree!

### ISSUE THREAD

**Issue by Alexey-Ost** [coppolaemilio/dialogic#325]
Possibility to use event name/tag/id for choose where to go in the dialog.

next(name/id) — makes transition and memorized previous position in stack (for return).
back() — makes return to 1 position in memory stack; back(name/id)  — return to name/id.

It's like 'push/pop' in _instead_engine_ — [https://github.com/instead-hub/instead/blob/master/doc/stead3-ru.md#%D0%B4%D0%B8%D0%B0%D0%BB%D0%BE%D0%B3%D0%B8](url)
That have cool and powerfull dialogs system!

**Comment by Alexey-Ost** [comment]
A few mockups:
![Снимок экрана от 2021-05-30 19-11-10](https://user-images.githubusercontent.com/1687903/120112579-19a23400-c17f-11eb-9029-de0c22e57bff.png)
![Снимок экрана от 2021-05-30 19-27-13](https://user-images.githubusercontent.com/1687903/120112591-21fa6f00-c17f-11eb-9626-4273967369cd.png)
![Снимок экрана от 2021-05-30 19-24-32](https://user-images.githubusercontent.com/1687903/120112607-2e7ec780-c17f-11eb-94c9-0c45af08f68d.png)
![Снимок экрана от 2021-05-30 19-09-47](https://user-images.githubusercontent.com/1687903/120112633-4c4c2c80-c17f-11eb-9534-36232c816260.png)
![Снимок экрана от 2021-05-30 19-37-23](https://user-images.githubusercontent.com/1687903/120112663-62f28380-c17f-11eb-8d01-187b18fe9222.png)

**Comment by Jowan-Spooner** [comment]
My proposal would be to add an "Anchor" event and the ability to select an anchor on the change timeline event. I think it could solve this. What do you think?

**Comment by Alexey-Ost** [comment]
Since I worked with the tag system in the dialogs "Instead engine" and it turned out to be very convenient (try it!), so I just try to give something similar in the 'Dialogic' system. Perhaps, "anchor" it is the same thing...

The bottom line is that you can move between phrases (up and down), go to previous point of choosing and choose a different path.

This makes a more non-linear conversation, but brings the player to a certain phrase and we have an desired result.

At the same time, we can go back and make a different choice. And you don't need to duplicate parts for different branches of the dialog.

 ::)

**Comment by coppolaemilio** [comment]
hello there! thanks for the issue, I will look deeper into that other `instead_engine ` system you linked. 
What I initially had in mind was adding a checkbox option to each of the choices inside a question to "go back to question". If checked, it would go back to the question and show the choices again for the user to pick a new one. Tags is something that I also was considering, but didn't get to go over them yet.

**Comment by Alexey-Ost** [comment]
> a checkbox option to each of the choices inside a question to "go back to question". If checked, it would go back to the question and show the choices again for the user to pick a new one.

Yes, this will be a good option if we need to stay inside the "question". And you don't need to go to some completely different part of the dialog after selecting.

**Comment by HarleyDavidson86** [comment]
Yes I also would really like this option "stay inside question"!
